sql/catalyst/src/main/scala/org/apache/spark/sql/types/DataType.scala

```diff
@@ -79,8 +80,12 @@ abstract class DataType extends AbstractDataType {
    * Check if `this` and `other` are the same data type when ignoring nullability
    * (`StructField.nullable`, `ArrayType.containsNull`, and `MapType.valueContainsNull`).
    */
-  private[spark] def sameType(other: DataType): Boolean =
-    DataType.equalsIgnoreNullability(this, other)
+  private[spark] def sameType(other: DataType, isCaseSensitive: Boolean = true): Boolean =
```

Comment:
maybe we should not consider field names in `sameType`, @gatorsmile what do you think?